explainers/hidden-content-explainer.md

```diff
@@ -71,196 +75,152 @@ currently possible.
+`scroll` events. It is already possible to snoop on find-in-page by creating a
 scrollable area containing every next character the user could type into
 find-in-page, listening to scroll events to see which caracter the user typed
 in, then prepending the new character to all of the next possible search terms.
 If the scrollable area is 1px tall or otherwise very small or hard to see, then
 the user may not be able to tell it is happening. This is demonstrated in
 [search-incremental.html](/resources/find-in-page/search-incremental.html).
 
-In order to mitigate the additional information that beforematch could expose to
-the page, we have added two constraints to beforematch:
-
-1. The beforematch event will only be fired on content in the DOM subtree of an
-   element with `content-visibility: hidden-matchable`. This removes the
-   information delta of find-in-page on normal pages without hidden find-in-page
-   snooping. Without this constraint, a page could listen to beforematch events
-   and get some idea of what you're searching for within the text already in the
-   page. Although a page could already try to do this by listening to scroll
-   events and guessing which element was scrolled to, the beforematch event
-   without this restriction is a stronger and more accurate signal.
-
-2. If the page fails to reveal the active match when the browser tries to scroll
-   to the active match after firing the beforematch event, the beforematch event
-   will be disabled for the remaining lifetime of the document. By "reveal," I
-   mean that the active match fulfills all of these requirements:
-   - The DOM range of the target match is not
-     [collapsed](https://dom.spec.whatwg.org/#range-collapsed), meaning that the
-     target match was not removed from the DOM.
-   - Neither the match nor any of its ancestors have `content-visibility:
-     hidden` or `content-visibility: hidden-matchable`.
-   - Neither the match nor the any of its ancestors have `display: none`.
-   - The `visibility` CSS property is `visibility: visible`.
-   This mitigation will prevent the page from building out a string of what the
-   user is searching for without having to reveal that string to the user
-   visually. Although this is already possible as shown in 
-   [search-incremental.html](/resources/find-in-page/search-incremental.html),
-   firing beforematch on hidden content that stays hidden would be harder for
-   the user to notice.
-
-As for the ScrollToTextFragment case, the page can already guess with reasonable
-certainty based on the scroll offset which text is already highlighted. In
-addition, ScrollToTextFragment can only occur once at the beginning of a
-document's load, so we don't have to worry about the page building out a search
-term since there will only be one possible beforematch event.
+In order to mitigate find-in-page snooping from both `scroll` events and
```

Comment:
I personally don't like the word "snooping" in this context, but it's up to you :)

Reply:
Sounds nefarious and informal. But it's fine, snooping it is!